src/features/farming/crops/components/Field.tsx

```diff
@@ -40,6 +43,36 @@ export const Field: React.FC<Props> = ({
   const clickedAt = useRef<number>(0);
   const field = game.context.state.fields[fieldIndex];
   const [showCropDetails, setShowCropDetails] = useState(false);
+  const [showRemoveWarning, setShowRemoveWarning] = useState(false);
+  const [lastCropRemovedTime, setLastCropRemovedTime] = useState<number | null>(
```

Comment:
I would suggest showing the modal everytime that you need to remove a crop. Since the main use case is fixing up a few errors (i.e. incorrectly planted crops), then we should design around this core focus.

There is quite a bit of extra logic to do the conditional rendering of the modal which adds a bit of bloat, maintenance and potential bugs in the future.

Reply:
> I would suggest showing the modal everytime that you need to remove a crop. Since the main use case is fixing up a few errors (i.e. incorrectly planted crops), then we should design around this core focus.

@0xsunflowerstudios This is definitely not a blocker for me - I am willing to change this if needed so we can get the feature out to the community :+1: . However, it's worth mentioning that the tech requirements of this feature went through a few iterations on my end with several community members. The feedback I have received by community has greatly indicated the desire to freely remove crops, without a warning modal constantly hounding the player.

> There is quite a bit of extra logic to do the conditional rendering of the modal which adds a bit of bloat, maintenance and potential bugs in the future.

@0xsunflowerstudios perhaps a refactor would be best here? Or alternatively, perhaps we create a new "Chore Request" to clean-up this logic after merge? I agree that this adds bloat, but I also wouldn't want to deny the community a feature that provides them value.